uaclient/apt.py

```diff
@@ -170,3 +174,54 @@ def remove_apt_list_files(repo_url, series):
     for path in find_apt_list_files(repo_url, series):
         if os.path.exists(path):
             os.unlink(path)
+
+
+def reconfigure_apt_sources(cfg=None, platform_info=None):
+    """Reconfigure apt sources for the given series.
+
+    Automatically setup an unauthenticated apt source for esm
+    production if not already present.
+    For each configured and enabled entitlement, update the sources files to
+    the new series.
+
+    @param platform_info: dict of platform information for testing
+    """
+    from uaclient import config
+    from uaclient import entitlements
+
+    if not platform_info:  # for testing
+        platform_info = util.get_platform_info()
+    if not cfg:  # for testing
+        cfg = config.UAConfig()
+
+    policy, _err = util.subp(['apt-cache', 'policy'])
+    for ent_cls in entitlements.ENTITLEMENT_CLASSES:
+        if not hasattr(ent_cls, 'repo_url'):
+            continue
+        repo_filename = ent_cls.repo_list_file_tmpl.format(
+            name=ent_cls.name, series=platform_info['series'])
+        repo_list_glob = ent_cls.repo_list_file_tmpl.format(
+            name=ent_cls.name, series='*')
+        repo_url = ent_cls.repo_url
+        if cfg.is_attached:
+            resource_cfg = cfg.entitlements.get(ent_cls.name, {})
+            directives = resource_cfg.get(
+                'entitlement', {}).get('directives', {})
+            repo_url = directives.get('aptURL')
+        enabled = bool(re.search(r'(?P<pin>(-)?\d+) %s' % repo_url, policy))
+        if not os.path.exists(repo_filename):
+            if ent_cls.name == 'esm':
+                if platform_info['series'] == 'trusty':
```

Comment:
Rather than hard-coding trusty, is there something we can check to ensure that series is supported?

Reply:
I take this back, this is used for first install, so we wouldn't know what was or wasn't supported. Ignore this :)

Reply:
right, the only type of system we'd be able to check supported/not is after upgrade from a previous ubuntu-advantage-tools v19 or later which had already been attached. Then we'd have entitlement access information for support levels.

Reply:
I've added a cfg.is_attached check where we loop over 'active' entitlements and re-enable that entitlement if the new series is reported as supported by ua-contracts